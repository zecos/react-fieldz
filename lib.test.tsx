import { test, /*describe, before, after, afterEach, beforeEach*/ } from 'tezt'
import { useFieldz } from './lib'
import expect from "expect"
import { render, fireEvent, waitForElement } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

const fieldProperties = {
  firstName: {
    errors: [],
    touched: false,
    pristine: true,
    value: ''
  },
  customField: {
    errors: [],
    touched: false,
    pristine: true,
    value: 'this is my init value'
  }
}

const camelToTitle = camelCase => camelCase
    .replace(/([A-Z])/g, match => ` ${match}`)
    .replace(/^./g, match => match.toUpperCase())
    .trim()

const TestForm = () => {
  const [{actions, fieldsState}, setFieldsState] = useFieldz(fieldProperties)
  const {setValue, setValues, setTouched, resetField, resetFields } = actions

  return (
    <form>
      {Object.entries(fieldsState)
        .map(([fieldName, {errors, value, touched, pristine}]) => (
          <div>
            {(touched && errors.length) ? <span className="input-error">{errors.map(err => <div>{err.toString()}</div>)}</span> : ""}
            <label htmlFor={fieldName}>{camelToTitle(fieldName)}</label>
            <input
              name={fieldName}
              value={value}
              onChange={e => setFieldsState(setValue(fieldName, e.target.value))}
              onBlur={_ => setFieldsState(setTouched(fieldName))}
            />
          </div>
        ))
      }
    </form>
  )
}
test("it works", () => {
  const { getByText, getByRole, getByLabelText } = render(<TestForm />)
  const input = getByLabelText("Custom Field")
  console.log(input.value)

})