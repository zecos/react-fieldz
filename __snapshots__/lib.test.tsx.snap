// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it works 1`] = `
<DocumentFragment>
  <form>
    <div>
      Error: []
    </div>
    <div>
      Error: ["Error: value must be hello!"]
    </div>
    <div>
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        aria-label="firstName"
        id="firstName"
        name="firstName"
        value="Zane"
      />
    </div>
    <div>
      <label
        for="customField"
      >
        Custom Field
      </label>
      <input
        aria-label="customField"
        id="customField"
        name="customField"
        value="helo"
      />
    </div>
  </form>
</DocumentFragment>
`;

exports[`it works 2`] = `
<DocumentFragment>
  <form>
    <div>
      Error: []
    </div>
    <div>
      Error: []
    </div>
    <div>
      <label
        for="firstName"
      >
        First Name
      </label>
      <input
        aria-label="firstName"
        id="firstName"
        name="firstName"
        value="Zane"
      />
    </div>
    <div>
      <label
        for="customField"
      >
        Custom Field
      </label>
      <input
        aria-label="customField"
        id="customField"
        name="customField"
        value="hello"
      />
    </div>
  </form>
</DocumentFragment>
`;
